devtools/gradle/gradle-extension-plugin/src/main/java/io/quarkus/extension/gradle/QuarkusExtensionPlugin.java

```diff
@@ -48,20 +48,19 @@ private void registerTasks(Project project, QuarkusExtensionConfiguration quarku
                 .getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME);
 
         TaskProvider<ExtensionDescriptorTask> extensionDescriptorTask = tasks.register(EXTENSION_DESCRIPTOR_TASK_NAME,
-                ExtensionDescriptorTask.class, task -> {
-                    JavaPluginConvention convention = project.getConvention().getPlugin(JavaPluginConvention.class);
-                    SourceSet mainSourceSet = convention.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
-                    task.setOutputResourcesDir(mainSourceSet.getOutput().getResourcesDir());
-                    task.setInputResourcesDir(mainSourceSet.getResources().getSourceDirectories().getAsPath());
-                    task.setQuarkusExtensionConfiguration(quarkusExt);
-                    task.setClasspath(runtimeModuleClasspath);
-                });
+                ExtensionDescriptorTask.class, quarkusExt);
+        extensionDescriptorTask.configure(task -> {
```

Comment:
You could even consider moving this into the `ExtensionDescriptorTask` constructor.